scripts/apidoc/moduleMethods.ts

```diff
@@ -38,7 +38,7 @@ export function processModuleMethods(project: ProjectReflection): PageIndex {
 }
 
 export function extractModuleName(module: DeclarationReflection): string {
-  return module.name.replace(/Module$/, '');
+  return (module.escapedName || module.name).replace(/Module$/, '');
```

Comment:
This change doesn't seem related to this PR. If so, how?

Reply:
The `name` attribute is derived/calculated by typedoc.
IMO it is a bug from typedoc, that it uses the alias as the main name (because it is exported first).
If we disable our formatter/linter and swap the order of our imports/exports, then typedoc picks the correct name.
To avoid that issue we use the name that typescript assigns to that type (`escapedName`).
(I don't know why it is called `escapedName` though.)
The `escapedName` can be null (says typescript), but it never is, so basically we always use the `escapedName` instead of `name`. The code is effectively like this:

````ts
return module.escapedName.replace(/Module$/, '');
````

This change is required for the docs to work/generate the correct api docs.

Reply:
I still don't see how this is related (and only now required) to renaming the `NameModule` to `PersonModule`. 
So, if the renaming would be reverted this line (`return module.name.replace(/Module$/, '');`) would break?

Reply:
I also had a deeper look into that. `escapedName` is marked as `@internal`, should really use it then?

Reply:
> I still don't see how this is related (and only now required) to renaming the `NameModule` to `PersonModule`.

This is required because typedoc doesn't assign the correct name.
The change is only relevant because we created an alias for the first time.
I don't consider this permanent code, maybe we can add a comment stating we should remove it later on.

> So, if the renaming would be reverted this line (return module.name.replace(/Module$/, '');return (module.escapedName || module.name).replace(/Module$/, '');) would break?

No, it doesn't. If you flip the imports/exports, then also name would show the correct value.

> I also had a deeper look into that. escapedName is marked as @internal, should really use it then?

Yes, the field is internal, but I couldn't find another way to achieve the same result.
(Aside from droping typedoc and using the typescript AST directly)
If you have an idea for a different fix, please tell me.
When I have some time, I will write a minimal reproduction and report this issue to typedoc.

We could write a test, to ensure that `faker.name` doesn't show up and `faker.person` does.

I don't think reversing the import/export order (and disabling the formatter/linter for that) is an acceptable alternative.

Reply:
Should we write instead:

````suggestion
const { name } = module;
if (name === 'NameModule') {
    return 'Person';
} else {
    return name.replace(/Module$/, '');
}
````

(not tested)

Reply:
This would be a potential solution, but looking forward to address->location rename we would need it for every alias
Your actual solution works and handle all the cases dynamically

Reply:
> Should we write instead:
> 
> ```ts
> const { name } = module;
> if (name === 'NameModule') {
>     return 'Person';
> } else {
>     return name.replace(/Module$/, '');
> }
> ```

The is makes in much easer for me to understand whats happening here.

Reply:
> This would be a potential solution, but looking forward to address->location rename we would need it for every alias Your actual solution works and handle all the cases dynamically

Sure, but we have it for the definitions proxy already.
See also my (future) comment below.